scipy/stats/_hypotests.py

```diff
@@ -139,6 +141,253 @@ def epps_singleton_2samp(x, y, t=(0.4, 0.8)):
+    -----
+    The Poisson distribution is commonly used to model many processes such as
+    transactions per user. Unlike the simpler C-test [1]_, which follows a
+    conditional distribution, the E-test is an unconditional test.
+
+    The E-test has been evaluated and determined to be more powerful than the
+    comparable C-test [1]_. The E-test can be said to be almost exact because
+    the test exceeds the nominal value by a negligible amount, whereas the
+    C-test's results are smaller than the nominal value.
+
+    References
+    ----------
+    .. [1]  Krishnamoorthy, K., & Thomson, J. (2004). A more powerful test for
+       comparing two Poisson means. Journal of Statistical Planning and
+       Inference, 119(1), 23-35.
+
+    .. [2]  Przyborowski, J., & Wilenski, H. (1940). Homogeneity of results in
+       testing samples from Poisson series: With an application to testing
+       clover seed for dodder. Biometrika, 31(3/4), 313-323.
+
+    Examples
+    --------
+    >>> from scipy import stats
+
+    Suppose that a gardener wishes to test the number of dodder seeds,
+    a weed, in a sack of clover seeds that they will buy from seed company.
+    A 100 gram sample is drawn from the sack prior to being shipped to the
+    gardener. The sample is analyzed, and it is found to contain no dodder
+    seeds; that is, count1 = 0. However, upon arrival, the gardener draws
+    another 100 gram sample from the sack. This time, three dodder seeds are
+    found in the sample; that is, count2 = 3. The user would like to know if
+    the difference between is significant and not due to chance. The null
+    hypothesis is that the variance in number of dodder seeds is merely due
+    to chance. The gardener decides that a significance level of 5% is
+    required to reject the null hypothesis in favor of the alternative.
+    (Przyborowski and Wilenski (1940) [2]_)
+
+    >>> res = stats.poisson_means_test(0, 100, 3, 100)
+    >>> res.statistic, res.pvalue
+    -1.7320508075688772, 0.08837900929018155
+
+    The p-value is .088, indicating a near 9% chance of observing a value of
+    the test statistic under the null hypothesis. This exceeds 5%, so the
+    gardener does not reject the null hypothesis as the difference cannot be
+    regarded as significant.
+
+    Examine the same data using the more common t-test:
+
+    >>> res = stats.ttest_ind_from_stats(mean1=0, std1=0, nobs1=100,
+    ...                                  equal_var=False, mean2=3.0/100,
+    ...                                  std2=np.sqrt(.03), nobs2=100)
+    >>> res.statistic, res.pvalue
+    -1.7320508075688772, 0.0863790757063214
+
+    Both t-test and the poisson mean test show similar p-values and do not
+    reject the null hypothesis. The power of the poisson mean test is
+    demonstrated with smaller sample sizes:
+
+    >>> res = stats.poisson_means_test(0, 10, 3, 10)
+    >>> res.statistic, res.pvalue
+    1.7320508075688772, 0.08837900929018155)
```

Comment:
If we want the refguide check to pass, it needs to be:
```suggestion
    (-1.7320508075688772, 0.08837900929018157)
```